www/BaseClass.js

```diff
@@ -1,181 +1,130 @@
+var VARS_FIELD = typeof Symbol === 'undefined' ? '__vars' + Date.now() : Symbol.for('vars');
+var SUBSCRIPTIONS_FIELD = typeof Symbol === 'undefined' ? '__subs' + Date.now() : Symbol.for('subscriptions');
 
-var _globalListeners = {};
+function BaseClass() {
+  this[VARS_FIELD] = {};
+  this[SUBSCRIPTIONS_FIELD] = {};
+  this.errorHandler = this.errorHandler.bind(this);
 
-function globalEventListener(e) {
-  var eventName = e.type;
-  if (!_globalListeners[eventName]) {
-    return;
-  }
-
-  var hashCode = e.myself.hashCode;
-  if (!hashCode || !_globalListeners[eventName][hashCode]) {
-    return;
-  }
-  var callbacks = _globalListeners[eventName][hashCode];
-  for (var i = 0; i < callbacks.length; i++) {
-    callbacks[i].listener(e);
-  }
+  Object.defineProperty(this, 'hashCode', { value: Math.floor(Date.now() * Math.random()) })
 }
 
-var BaseClass = function() {
-  var self = this;
-  var _vars = {};
+BaseClass.prototype = {
+  empty: function() {
+    var vars = this[VARS_FIELD];
 
-  var hashCode = Math.floor(Date.now() * Math.random());
+    Object.keys(vars).forEach(function(name) {
+      vars[name] = null;
+    });
+  },
+
+  get: function(key) {
+    return this[VARS_FIELD].hasOwnProperty(key) ? this[VARS_FIELD][key] : null;
```

Comment:
This should be `return this[VARS_FIELD].hasOwnProperty(key) ? this[VARS_FIELD][key] : undefined`.
The null === undefined always false.